### DISCUSSION TITLE
Help debugging memory leak

### DISCUSSION THREAD

**Discussion by Eloitor** [marimo-team/marimo#1108] [Q&A]
I'm using marimo to run sagemath. The notebooks work fine, but when I edit them sometimes I get mysterious memory leaks. 

I'd like to get a minimal example of this memory leak without using marimo, so I can report this problem to sagemath.

Using marimo I found this minimal example:
```
from sage.all import QuadraticField
F = QuadraticField(5)
F.prime_above(2) 
```

[pari_leak.py.txt](https://github.com/marimo-team/marimo/files/14946253/pari_leak.py.txt)

After editing the cell and running again, it crashes the python kernel:

[log.txt](https://github.com/marimo-team/marimo/files/14945414/log.txt)
[log2.txt](https://github.com/marimo-team/marimo/files/14946203/log2.txt)

Sometimes I get this
![imatge](https://github.com/marimo-team/marimo/assets/17795572/4d29dc90-1b6c-4770-8059-6a056a445748)

**Comment by akshayka** [comment]
Sorry, to clarify, do you think this is a leak in marimo or in SageMath?

**Reply by Eloitor** [reply]
I think it is a leak in Sagemath, but I'm unable to reproduce it without marimo...

**Comment by Eloitor** [comment]
![imatge](https://github.com/marimo-team/marimo/assets/17795572/cf2b7acc-38d4-4fef-ad6f-399c69196c47)

I got a segmentation fault. 

I wrote this in a single cell and run it:
```
from sage.all import QuadraticField
F = QuadraticField(5)
F.prime_above(2) 
```

Then I deleted the last line and run it again, getting a segmentation fault. 

I'm using
```
[experimental]
execution_type = "strict"
```

**Reply by akshayka** [reply]
cc @dmadisetti

**Reply by dmadisetti** [reply]
That's exciting. I'll try this out, strict mode cannot manage FFI memory, so something might be happening under the hood. 

Does this happen only in strict mode?

**Reply by dmadisetti** [reply]
I do get the crash, but not consistently / reliably. I don't know if this is strict mode specific either since I got the crash after I turned strict mode off.

Seemed a little hit or miss, sometimes restarting the kernel was fine and I couldn't trigger the response at all. Other times crashed first go. Seems like we need someone with an idea of sage internals to give some insight

**Reply by dmadisetti** [reply]
Notebook I was using for reference

[Sage.md](https://github.com/user-attachments/files/15930773/Sage.md)
